lib/cve.py

```diff
@@ -1,39 +1,105 @@
 #!/usr/bin/env python
 """
-python lib/get_cve.py \
+python lib/cve.py \
     tenant \
-    --git https://gitlab.cee.redhat.com/gnecasov/container-errata-templates.git \
-    --branch main \
-
-    git clone https://github.com/openshift/contra-lib.git --branch master --depth 1 "/tmp/1"
-
+    --release release.yaml \
+    --previousRelease previous_release.yaml 
 """
 
+
 import argparse
 import os
 import tempfile
 import re
 import subprocess
+import json
 
 pattern = r'CVE-\d+-\d+|CVE-\d+'
 
-
 def find_cve():
+    file_not_exists = 0
     parser = argparse.ArgumentParser()
     parser.add_argument(
             "mode",
             choices=["managed", "tenant"],
             help="Mode in which the script is called. It does not have any impact for this script.")
-    parser.add_argument("--git", required=True, help="SSH clone string for a git repository")
-    parser.add_argument("--branch", required=True, help="Branch name to be cloned, it can be a branch or a SHA.")
+    parser.add_argument('-r', '--release', help='Path to current release file', required=True)
+    parser.add_argument('-p', '--previousRelease', help='Path to previous release file', required=True)
     args = vars(parser.parse_args())
-    cves = git_log_titles(args['git'], args['branch'])
-    return create_json_record(cves, "my-component")
+
+    if (not os.path.isfile(args['release'])):
+        print(f"ERROR: Path to release file {args['release']} doesn't exists")
+        file_not_exists = 1
+    if (not os.path.isfile(args['previousRelease'])):
+        print(f"ERROR: Path to previousRelease file {args['previousRelease']} doesn't exists")
+        ile_not_exists = 1
+    if (file_not_exists):
+        exit(1)
+
+    return(components_info(args['release'], args['previousRelease']))
+
+
+def read_json(file):
+    data = {}    
+
+    if  os.path.getsize(file) > 0:
+        with open(file, 'r') as f:
+            data = json.load(f)
+        return data
+
+
+def get_component_names(data_list):
+    component_list = []
+    for component_info in data_list:
+        for key, value in component_info.items():
+            if (key == "name"):
+                component_list.append(value)
+    return component_list
 
 
-def git_log_titles(git_url, branch):
+def get_component_detail(data_list, component):
+    found = 0
+    for component_info in data_list:
+        for key, value in component_info.items():
+            if (key == "name") and (value == component):
+                found = 1
+            if (key == "source"):
+                return(value["git"]["url"],value["git"]["revision"])
+    return([])
+
+
+def components_info(release, previousRelease):
+    current_component_list = []
+    prev_component_list = []
+    cves = {}
+    data_release = read_json(release)
+    data_prev_release = read_json(previousRelease)
+
+    if data_release == None:
```

Comment:
You don't need to compare it to `None`, `if data_release` does the same job. Read more here: https://docs.python.org/3/library/stdtypes.html#truth-value-testing